### PR TITLE
Merge pull request #658 from iKostanOrg/master

### PR DESCRIPTION
**Pull request by ikostan**
Merge from master

## Summary by Sourcery

Refactor test suites to use parameterized tests and add more test cases to improve code quality and coverage. Update CI workflows to use the correct branch name.

CI:
- Update CI workflows to use the `kyu3` branch instead of `'kyu3'`.

Tests:
- Refactor test cases to use parameterized tests for improved readability and maintainability.
- Add more test cases to cover edge cases and improve test coverage for the Battleship field validator, Line Safari, Spiralize, Calculator, and Rail Fence Cipher problems.

### CODE REVIEW

**Comment by ikostan** [comment]
@sourcery-ai review

File: kyu_3/make_spiral/test_spiralize.py [hunk 1]
```diff
@@ -32,7 +33,43 @@
...
+             [1, 1, 1, 1, 1, 1, 1, 1]]),
+        (9, [[1, 1, 1, 1, 1, 1, 1, 1, 1],
+             [0, 0, 0, 0, 0, 0, 0, 0, 1],
+             [1, 1, 1, 1, 1, 1, 1, 0, 1],
+             [1, 0, 0, 0, 0, 0, 1, 0, 1],
+             [1, 0, 1, 1, 1, 0, 1, 0, 1],
+             [1, 0, 1, 0, 0, 0, 1, 0, 1],
+             [1, 0, 1, 1, 1, 1, 1, 0, 1],
+             [1, 0, 0, 0, 0, 0, 0, 0, 1],
+             [1, 1, 1, 1, 1, 1, 1, 1, 1]])])
```
[the remark this answers is not in the document]
Comment (ikostan) [reply kyu_3/make_spiral/test_spiralize.py:71]: Because of the edge-cases for tiny spirals, the size will be at least `5`.

File: kyu_3/rail_fence_cipher_encoding_and_decoding/test_decoding.py [hunk 2]
```diff
@@ -33,7 +34,13 @@
 class DecodingTestCase(unittest.TestCase):
     """Testing Decoding functionality."""
 
-    def test_decoding(self):
+    @parameterized.expand([
+        ("H !e,Wdloollr", 4, "Hello, World!"),
+        ("WECRLTEERDSOEEFEAOCAIVDEN", 3, "WEAREDISCOVEREDFLEEATONCE"),
+        ("", 3, ""),
+        ("WEAREDISCOVEREDFLEEATONCE", 10, "WADCEDETNECOEFROIREESVELA"),
+        ("WEAREDISCOVEREDFLEEATONCE", 9, "WADCEDETCOEFROIREESVELANE")])
```
[the remark this answers is not in the document]
Comment (ikostan) [reply kyu_3/rail_fence_cipher_encoding_and_decoding/test_decoding.py:42]: For both encoding and decoding, assume number of rails >= 2 and that
passing an empty string will return an empty string.

File: kyu_3/rail_fence_cipher_encoding_and_decoding/test_encoding.py [hunk 3]
```diff
@@ -33,7 +34,16 @@
...
-    def test_encoding(self):
+    @parameterized.expand([
+        ("WEAREDISCOVEREDFLEEATONCE", 3, "WECRLTEERDSOEEFEAOCAIVDEN"),
+        ("Hello, World!", 3, "Hoo!el,Wrdl l"),
+        ("Hello, World!", 4, "H !e,Wdloollr"),
+        ("", 3, ""),
+        ('WEAREDISCOVEREDFLEEATONCE', 4, 'WIREEEDSEEEACAECVDLTNROFO'),
+        ('WEAREDISCOVEREDFLEEATONCE', 6, 'WVTEOEAOACRENRSEECEIDLEDF'),
+        ('WEAREDISCOVEREDFLEEATONCE', 10, 'WEEEAALTRFOEDNDECIRESECVO'),
+        ("WEAREDISCOVEREDFLEEATONCE", 5, 'WCLEESOFECAIVDENRDEEAOERT')])
```
[the remark this answers is not in the document]
Comment (ikostan) [reply kyu_3/rail_fence_cipher_encoding_and_decoding/test_encoding.py:45]: For both encoding and decoding, assume number of rails >= 2 and that
passing an empty string will return an empty string.
